catalyst/contrib/nn/criterion/triplet.py

```diff
@@ -257,4 +261,75 @@ def forward(self, embeddings_pred, embeddings_true):
         return loss
 
 
-__all__ = ["TripletLoss", "TripletPairwiseEmbeddingLoss"]
+class TripletMarginLossWithSampling(nn.Module):
+    """
+    This class combains in-batch sampling of triplets and
+    default TripletMargingLoss from PyTorch.
+    """
+
+    def __init__(
+        self, margin: float, sampler_inbatch: si.InBatchTripletsSampler
+    ):
+        """
+        Args:
+            margin: margin value
+            sampler_inbatch: sampler for forming triplets inside the batch
+        """
+        super().__init__()
+        self._sampler_inbatch = sampler_inbatch
+        self._triplet_margin_loss = TripletMarginLoss(margin=margin)
+
+    @staticmethod
+    def _prepate_labels(labels: Union[Tensor, List[int]]) -> List[int]:
+        """
+        This function allows to work with 2 types of indexing:
+        using a integer tensor and a list of indices.
+
+        Args:
+            labels: labels of batch samples
+
+        Returns:
+            labels of batch samples in the aligned format
+        """
+        if isinstance(labels, Tensor):
+            labels = labels.squeeze()
+            assert (len(labels.shape) == 1) and (
+                labels.dtype in [short, tint, long]
+            ), "Labels cannot be interpreted as indices."
+            labels_list = labels.tolist()
+
+        elif isinstance(labels, list):
+            labels_list = labels.copy()
+
+        else:
+            raise TypeError(f"Unexpected type of labels: {type(labels)}).")
+
+        return labels_list
+
+    def forward(
+        self, features: Tensor, labels: Union[Tensor, List[int]]
+    ) -> Tensor:
+        """
+        Args:
+            features: features with the shape of [batch_size, features_dim]
+            labels: labels of samples having batch_size elements
+
+        Returns: loss value
+
+        """
+        labels_list = self._prepate_labels(labels)
+
+        features_a, features_p, features_n = self._sampler_inbatch.sample(
```

Comment:
could we use full names? `features_anchor`, `features_positive`, `features_negative`?

Reply:
ok